### ISSUE TITLE
No option named --no-codesign

### ISSUE THREAD

**Issue by taaaf11** [flet-dev/flet#2340]
**Description**
The app builds correctly for android, with `flet build apk`, but when building for iphone `flet build ipa`, it fails with error in `flutter`, `no option named --no-codesign`.

**Describe the results you received:**
Error

**Describe the results you expected:**
Expected: Application package built for iphone.

**Additional information you deem important (e.g. issue happens only occasionally):**
It only happens with `flet build ipa`.

**Flet version (`pip show flet`)**:
```
Name: flet
Version: 0.18.0
Summary: Flet for Python - easily build interactive multi-platform apps in Python
Home-page:
Author: Appveyor Systems Inc.
Author-email: hello@flet.dev
License: Apache-2.0
Location: C:\Users\Altaaf\PycharmProjects\TaDo\.env\Lib\site-packages
Requires: cookiecutter, flet-runtime, packaging, qrcode, watchdog, websocket-client, websockets
Required-by:
```

**Operating system**:

`Windows 10 22H2 Build 19045.3803`

Here is the output with `-vv` option:
```
Flutter bootstrap directory: C:\Users\Altaaf\AppData\Local\Temp\flet_flutter_build_FpeIaOFGqa
Creating Flutter bootstrap project...OK
Customizing app icons and splash images...OK
Generating app icons...Resolving dependencies in C:\Users\Altaaf\AppData\Local\Temp\flet_flutter_build_FpeIaOFGqa... (5.8s)
Got dependencies in C:\Users\Altaaf\AppData\Local\Temp\flet_flutter_build_FpeIaOFGqa.
Building package executable... (8.8s)
Built flutter_launcher_icons:flutter_launcher_icons.
  ════════════════════════════════════════════
     FLUTTER LAUNCHER ICONS (v0.13.1)
  ════════════════════════════════════════════

• Creating default icons Android
• Overwriting the default Android launcher icon with a new icon
• Overwriting default iOS launcher icon with new icon
Creating Icons for Web...
Creating Icons for Windows...
Creating Icons for MacOS...

✓ Successfully generated launcher icons
OK
Generating splash screens...Building package executable... (8.5s)
Built flutter_native_splash:create.
[iOS] Creating  images
[iOS] Creating dark mode  images
[iOS] Updating ios/Runner/Info.plist for status bar hidden/visible
╔════════════════════════════════════════════════════════════════════════════╗
║                       NEED A GREAT FLUTTER DEVELOPER?                      ║
╠════════════════════════════════════════════════════════════════════════════╣
║                                                                            ║
║   I am available!  Find me at https://www.linkedin.com/in/hansonjon/       ║
║                                                                            ║
╚════════════════════════════════════════════════════════════════════════════╝

✅ Native splash complete.
Now go finish building something awesome! 💪 You rock! 🤘🤩
Like the package? Please give it a 👍 here: https://pub.dev/packages/flutter_native_splash

OK
Packaging Python app...Building package executable... (4.7s)
Built serious_python:main.
Running package command
Creating asset directory: C:\Users\Altaaf\AppData\Local\Temp\flet_flutter_build_FpeIaOFGqa\app
Copying Python app from C:\Users\Altaaf\Documents\TaDo to C:\Users\Altaaf\AppData\Local\Temp\serious_python_temp244884bf
Installing dependencies [flet-embed] with pip command to C:\Users\Altaaf\AppData\Local\Temp\serious_python_temp244884bf\__pypackages__
Extracting Python distributive from C:\Users\Altaaf\AppData\Local\Temp\cpython-3.11.6+20231002-x86_64-pc-windows-msvc-shared-install_only.tar.gz to C:\Users\Altaaf\AppData\Local\Temp\hostpython3.11_d22731a5
Compiling Python sources at C:\Users\Altaaf\AppData\Local\Temp\serious_python_temp244884bf
Delete unnecessary files with extensions: [.py, .c, .h, .typed, .exe, .so, .a, .pdb, .pyd, .dll]
Delete unnecessary files and directories: [__pycache__, bin]
Creating app archive at C:\Users\Altaaf\AppData\Local\Temp\flet_flutter_build_FpeIaOFGqa\app/app.zip from C:\Users\Altaaf\AppData\Local\Temp\serious_python_temp244884bf
Deleting temp directory C:\Users\Altaaf\AppData\Local\Temp\serious_python_temp244884bf
Deleting Python directory C:\Users\Altaaf\AppData\Local\Temp\hostpython3.11_d22731a5
OK
Building .ipa bundle for iOS...['C:\\Flutter\\flutter\\bin\\flutter.BAT', 'build', 'ipa', '--no-codesign', '--build-number', '1', '--build-name', '3.0.0']
Could not find an option named "no-codesign".


Run 'flutter -h' (or 'flutter <command> -h') for available flutter commands and options.
Deleting Flutter bootstrap directory C:\Users\Altaaf\AppData\Local\Temp\flet_flutter_build_FpeIaOFGqa
Error building Flet app - see the log of failed command above.
```

**Comment by FeodorFitsner** [comment]
What Flutter version do you have?

**Comment by taaaf11** [comment]
`flutter --version`:
```
Flutter 3.16.5 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 78666c8dc5 (3 weeks ago) • 2023-12-19 16:14:14 -0800
Engine • revision 3f3e560236
Tools • Dart 3.2.3 • DevTools 2.28.4
```

**Comment by FeodorFitsner** [comment]
`Could not find an option named "no-codesign".` worries me (notice `no-codesign` without `--`).
But anyway you can't build iOS apps on Windows.

**Comment by taaaf11** [comment]
May this is of some help [flutter issue](https://github.com/flutter/flutter/issues/88569#issuecomment-903633347) ...

**Comment by FeodorFitsner** [comment]
Saw that, thanks. I can re-produce the issue when running the command on Windows, but on macOS it works without issues. But as I noted earlier you can build iOS apps on macOS only - [check this matrix](https://flet.dev/docs/guides/python/packaging-app-for-distribution#build-platform-matrix).

I think, to avoid future confusions we need to add check for OS when a certain `flet build` command is run.

**Comment by taaaf11** [comment]
Okay, thank you

**Comment by taaaf11** [comment]
I agree

**Comment by FeodorFitsner** [comment]
Closing in favour of https://github.com/flet-dev/flet/issues/2342
